arkouda/io.py

```diff
@@ -631,6 +675,89 @@ def read_parquet(
+    datasets: Optional[Union[str, List[str]]] = None,
+    column_delim: str = ",",
+    allow_errors: bool = False,
+) -> Union[
+    pdarray,
+    Strings,
+    arkouda.array_view.ArrayView,
+    Mapping[str, Union[pdarray, Strings, arkouda.array_view.ArrayView]],
+]:
+    """
+    Read CSV file(s) into Arkouda objects. If more than one dataset is found, the objects
+    will be returned in a dictionary mapping the dataset name to the Arkouda object
+    containing the data. If the file contains the appropriately formatted header, typed
+    data will be returned. Otherwise, all data will be returned as a Strings object.
+
+    Parameters
+    -----------
+    filenames: str or List[str]
+        The filenames to read data from
+    datasets: str or List[str] (Optional)
+        names of the datasets to read. When `None`, all datasets will be read.
+    column_delim: str
+        The delimiter for column names and data. Defaults to ",".
+    allow_errors: bool
+        Default False, if True will allow files with read errors to be skipped
+        instead of failing.  A warning will be included in the return containing
+        the total number of files skipped due to failure and up to 10 filenames.
+
+    Returns
+    --------
+    pdarray, Strings or Mapping {dset_name: obj} where obj is a pdarray or Strings.
+
+    Raises
+    ------
+    ValueError
+        Raised if all datasets are not present in all parquet files or if one or
+        more of the specified files do not exist
+    RuntimeError
+        Raised if one or more of the specified files cannot be opened.
+        If `allow_errors` is true this may be raised if no values are returned
+        from the server.
+    TypeError
+        Raised if we receive an unknown arkouda_type returned from the server
+
+    See Also
+    ---------
+    to_csv
+
+    Notes
+    ------
+    - CSV format is not currently supported by load/load_all operations
+    - The column delimiter is expected to be the same for column names and data
+    - Be sure that column delimiters are not found within your data.
+    - All CSV files must delimit rows using newline (`\n`) at this time.
+    - Unlike other file formats, CSV files store Strings as their UTF-8 format instead of storing
+    bytes as uint(8).
+    """
+    if isinstance(filenames, str):
+        filenames = [filenames]
+
+    if isinstance(datasets, str):
+        datasets = [datasets]
+    elif datasets is None:
+        datasets = get_columns(filenames, col_delim=column_delim, allow_errors=allow_errors)
+
+    rep_msg = generic_msg(
+        cmd="readcsv",
+        args={
+            "filenames": filenames,
+            "nfiles": len(filenames),
+            "datasets": datasets,
+            "num_dsets": len(datasets),
+            "col_delim": column_delim,
+            "allow_errors": allow_errors,
+        },
+    )
+    rep = json.loads(rep_msg)  # See GenSymIO._buildReadAllMsgJson for json structure
+    _parse_errors(rep, False)
```

Comment:
@pierce314159 - This function is only intended to produce the warning to the user when `allow_errors` is `True`.  I am not sure I am fully following what your are saying since the variables generated are needed to produce the warning message. Maybe we should discuss this more today.

Reply:
@pierce314159 - misunderstood your comment. The `False` should be `allow_errors`. It is updated now. Good catch.